components/PostForm/PostForm.js

```diff
@@ -0,0 +1,48 @@
+import styled from "styled-components";
+
+export function Postform() {
+  function handleClick(event) {
+    event.preventDefault();
+    document.getElementById("display").innerHTML =
+      document.getElementById("inputText").value;
+  }
+
+  return (
+    <Post>
+      <div>
+        <label>Post</label>
+        <input type="text" id="inputText" required name="text" />
+        <button type="submit" onClick={handleClick}>
```

Comment:
Rather than listening for the `click` event here, try to use the `submit` event (`onSubmit`) on the form itself.
You'll need to use `event.preventDefault()`.

It's cleaner because it catches all means by which the user might submit you form.

Reply:
Also the submit event will only be triggered for valid submits:

You are using `required` on the text input. Thus, when you listen to the button click, it'll be called no matter if there is a value in the input. Using the submit event, your handler function will only be called once the form is valid.